### ISSUE TITLE
Define default title and also titles for specific languages

### ISSUE THREAD

**Issue by heikobornholdt** [infopark/reactor#15]
With the Tcl API I can create a new attribute and specify values for attribute `title` and `title.en` (or `de`, `fr` etc.).
How can I do this with a reactor migration?

**Comment by tomaszp-infopark** [comment]
Sorry, for the (extremely) delayed answer.

When you set the language, via the migration `set :title, "Default Title` or Tcl `attribute withName test1 set title "default title"` in just sets `title.de`.

See for example:
```
root@localhost:5782 test(CM)> attribute create name test1  type string
root@localhost:5782 test(CM)> attribute withName test1 set title "default title"
root@localhost:5782 test(CM)> attribute withName test1 description
{
    "title.de" = "default title";
    editField = {
        length = 65;
        type = textfield;
    };
    isSearchableInCM = 1;
    isSearchableInTE = 1;
    name = test1;
    title = "default title";
    type = string;
}
```

There is no difference between a migration and Tcl even in case of multiple languages:

```ruby
class WithMigrations < Thorium::Migration
  def self.up
    create_attribute name: 'with_migrations', type: 'string' do
      set :title, {"German Title"=>{:lang=>:de}, "English Title"=>{:lang=>:en}}
    end
  end

  def self.down
  end
end
```
```
```
root@localhost:5782 test(CM)> attribute withName with_migrations description
{
    "title.de" = "German Title";
    "title.en" = "English Title";
    editField = {
        length = 65;
        type = textfield;
    };
    isSearchableInCM = 0;
    isSearchableInTE = 0;
    name = "with_migrations";
    title = "German Title";
    type = string;
}
```

```
root@localhost:5782 test(CM)> attribute create name with_tcl type string
with_tcl
root@localhost:5782 test(CM)> attribute withName with_tcl set title "German Title"
root@localhost:5782 test(CM)> attribute withName with_tcl set title.en "English Title"
root@localhost:5782 test(CM)> attribute withName with_tcl description
{
    "title.de" = "German Title";
    "title.en" = "English Title";
    editField = {
        length = 65;
        type = textfield;
    };
    isSearchableInCM = 1;
    isSearchableInTE = 1;
    name = "with_tcl";
    title = "German Title";
    type = string;
}
```

**Comment by tomaszp-infopark** [comment]
TL;DR I don't think a "default" title exists, instead `title` is an alias for `title.de`

**Comment by heikobornholdt** [comment]
Thanks. You're right!
